### ISSUE TITLE
Don't die when pruning a snapshot fails

### ISSUE THREAD

**Issue by itpflop** [jimsalterjrs/sanoid#149]
I'm using ZFS with NFS. After a snapshot is accessed via NFS it cannot be unmounted on the server anymore (see https://github.com/zfsonlinux/zfs/issues/6000). Unfortunately in this case sanoid fails:

```
cannot destroy snapshot bigwork/test@autosnap_2017-09-12_14:00:01_hourly: dataset is busy
could not remove bigwork/test@autosnap_2017-09-12_14:00:01_hourly : 256 at /usr/local/sbin/sanoid line 227.
```

I would like sanoid to remove as many old snapshots as possible instead of failing at the first non-removable snapshot. Then at least some of the snapshots are removed regularly. The rest would be purged after a reboot of the server.

If there are no technical reasons against this, a command line option to enable this behaviour would be perfect, I think.

**Comment by jimsalterjrs** [comment]
Changing this from **die** to **warn** seems perfectly reasonable.  I'll patch it now.

**Comment by jimsalterjrs** [comment]
fixed in 1.4.17.

**Comment by itpflop** [comment]
Perfect, thank you!
